src/index.ts

```diff
@@ -489,6 +559,45 @@ export function HTPipe(...objs: any[]) {
         : isHasSanitizeBody(right)
         ? { sanitizeBody: right.sanitizeBody }
         : {}) as PipedSanitizeBody<any, any>),
+      ...((isHasPreAuthorize(left) && isHasPreAuthorize(right)
+        ? {
+            preAuthorize: (context: any) => {
+              const leftOut = left.preAuthorize(context);
+              const leftPassed = authorizationPassed(
+                leftOut as boolean | object
```

Comment:
this is needed in few places because of this implementation leftOut becomes unknown so we can't use unknown for authorizationPassed function and object spread

Reply:
- [ ] footgun: we should change it below to if(!leftPassed || (typeof leftPassed == "object" && Object.keys(leftPassed).length===0)

Reply:
nvm :D i can't wait for this to be done